### ISSUE TITLE
Q: Running in virtual folder/context (under IIS + rewrite module)

### ISSUE THREAD

**Issue by evil-shrike** [janeczku/calibre-web#490]
Hi!
I'm trying to run calibre-web under IIS with RewriteModule. I want the ebook server to be running under /ebooks vitrual folder.

In my web.config I have rewrite rule:
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <directoryBrowse enabled="true" />
        <rewrite>
            <rules>
                <rule name="ReverseProxyInboundRule1" stopProcessing="true">
                    <match url="(.*)" />
                    <action type="Rewrite" url="http://1.2.3.4:8083/{R:1}" logRewrittenUrl="true" />
                    <serverVariables>
                        <set name="X-SCRIPT-NAME" value="/ebooks" />
                    </serverVariables>
                </rule>
            </rules>

```
But when I open the homepage it contains links from the root ("/") and they don't work.
How to tell calibre-web that it's running in /app context ?

![image](https://user-images.githubusercontent.com/1203230/38698172-d1339f78-3e9c-11e8-9fc5-0fe8d7ebf6b1.png)

p.s.
I had calibre-server.exe running in the same configuration with cmd:
```
calibre-server.exe"  --listen-on 1.2.3.4 --port 8080 --log log.txt --access-log log-access.txt --url-prefix /ebooks --enable-local-write --disable-use-bonjour --userdb users.sqlite
```

**Comment by evil-shrike** [comment]
I tried to add outbound rewrite rule to replace all links. But unfortunately IIS ReWrite module can't work with compression so I have to disable it via clearing Content-Type header but this breaks pdf reading.

```xml
            <rules>
                <rule name="ReverseProxyInboundRule1" stopProcessing="true">
                    <match url="(.*)" />
                    <action type="Rewrite" url="http://1.2.3.4:8083/{R:1}" logRewrittenUrl="true" />
                    <serverVariables>
                        <set name="X-SCRIPT-NAME" value="/ebooks" />
                        <set name="HTTP_X_ORIGINAL_ACCEPT_ENCODING" value="{HTTP_ACCEPT_ENCODING}" />
                        <set name="HTTP_ACCEPT_ENCODING" value="" />
                    </serverVariables>
                </rule>
            </rules>
            <outboundRules>
                <rule name="ReverseProxyOutboundRule1" preCondition="ResponseIsHtml1">
                    <match filterByTags="A, Area, Base, Form, Frame, Head, IFrame, Img, Link, Script" pattern="^/(.*)" />
                    <action type="Rewrite" value="/ebooks/{R:0}" />
                </rule>
                <preConditions>
                    <preCondition name="ResponseIsHtml1">
                        <add input="{RESPONSE_CONTENT_TYPE}" pattern="^text/html" />
                    </preCondition>
                </preConditions>
            </outboundRules>

```

**Comment by OzzieIsaacs** [comment]
I don't konw the config of IIS. I can help you only with one information:
1:) Yes, calibre-web doesn't support a config option for defining a path prefix, to my knowledge it should be always a without such an option.

One idea would be hardcode the path you need in the template files:
transform: 
<script src="{{ url_for('static', filename='js/libs/viewer.js') }}"></script>
to:
<script src="/yourpath/static/js/libs/viewer.js"></script>

What calibre-web is looking for is X-SCRIPT-NAME and X-SCRIPT-FORWARDED-SERVER entries in the header, so if you set these properly, calibre-web should be able to enter the correct path names into the html files in the responses.
 
![unbenannt](https://user-images.githubusercontent.com/6438895/38753432-aacb2488-3f5e-11e8-9a88-7e3ea062385a.png)

**Comment by evil-shrike** [comment]
I'm not sure I understand completely.

Ok, calibre-web has no an option for static configuration of path/context.
But README contains samples for reverse proxy configurations with ngix and apache - https://github.com/janeczku/calibre-web#reverse-proxy.
It's seems this's what I need.
In Apache sample there're only two headers are set: X-SCRIPT-NAME  and X-SCHEME. As I use http it's only one for me. And I do set it in IIS rewrite config.
```
<set name="X-SCRIPT-NAME" value="/ebooks" />
```

But it seems that it has no effect at all.

As for templates do you suggest hardcoding path in template as an addition to X-SCRIPT-NAME or as an alternative?

**Comment by evil-shrike** [comment]
So in general I should simplify my question - how to run calibre-web behind a reverse proxy in a context/path (like localhost/calibre)?

**Comment by OzzieIsaacs** [comment]
Problem is solved, see #755
